ports/sdl2-mixer/portfile.cmake

```
@@ -1,60 +1,36 @@
-set(SDL2_MIXER_VERSION 2.0.4)
-
-vcpkg_download_distfile(ARCHIVE
-    URLS "https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-${SDL2_MIXER_VERSION}.zip"
-    FILENAME "SDL2_mixer-${SDL2_MIXER_VERSION}.zip"
-    SHA512 359b4f9877804f9c4b3cb608ca6082aab684f07a20a816ab71c8cdf85d26f76d67eeb5aee44daf52b7935d82aa3b45941f8f53f07ca3dd5150d6c58ed99e1492
-)
-
-vcpkg_extract_source_archive_ex(
+vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
-    REF ${SDL2_MIXER_VERSION}
-    PATCHES
-        fix-featurempg123.patch
+    REPO libsdl-org/SDL_mixer
+    REF release-2.6.1
```

```suggestion
    REF 1cf51b6e16552e4118d90799ef752f7644efb9fd #release-2.6.1
```
Using a commit number is more recognizable and reliable

---

> Using a commit number is more recognizable

Opinion. I disagree. In fact you have to write separately which release it is meant to be, and you have to check it online.

>  and reliable

Reliable for every good project which doesn't modify tags.

---

I also believe this is safe enough since we have the SHA512 hash, but more readable. @FrankXie05 @dg0yt So what should I do? :)

---

> * Even stable releases have the potential to change content and commit

Technically yes. Practically, no for release tags. vcpkg is not the only user who would complain massively. And it is guarded by SHA512 here.

> * The determined commit number can accurately locate the last commit modified by our current version. Quickly find and understand what has changed in versions when there is a problem with a port.

This works with tags as well as with commits.

> * Should be changed to commit number

If you insist on this change, then I must insist on making this a published policy.
I had the same discussion some months ago, and the result was that both is acceptable.

---

No, you may have misunderstood. As you said, we accept both, but we want to use commit as much as possible, just like `license`.  :)

---

This means we will change back and forth, depending on current contributor.

---

No, we will add it when we update the port version (not mandatory), not as a separate request

---

> No, we will add it when we update the port version (not mandatory), not as a separate request

Now you misunderstood. I was saying that I will use tags where possible when I update ports. Even if there was a commit before.